fetch.js

```diff
@@ -217,6 +217,11 @@
 
   function Request(url, options) {
     options = options || {}
+
+    if (url && url.match(/^([\w-]+:)?\/\/[^\/]+@/)) {
```

Comment:
Looks alright now. Note that I'm not advocating use of regex to parse URLs in generally. For instance in cases when we had to detect cross-origin URLs in other libraries, using URL (or at least `HTMLAnchorElement#href` and related properties) was adamant to security (/cc @mastahyeti). However in this case I think it's relatively trivial to detect whether username or password have been specified, and raise an error appropriately. The worst thing that can happen is that this check fails and the XHR request made, but with user/password ignored because XHR never reads them from the URL string.